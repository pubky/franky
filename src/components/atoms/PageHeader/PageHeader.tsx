import * as Libs from '@/libs';
import * as Atoms from '@/atoms';

interface PageHeaderProps {
  children: React.ReactNode;
  className?: string;
  'data-testid'?: string;
}

export function PageHeader({
  children,
  className,
  'data-testid': dataTestId,
  ...props
}: PageHeaderProps & React.HTMLAttributes<HTMLDivElement>) {
  return (
    <Atoms.Container
      className={Libs.cn('gap-3', className)}
      data-testid={dataTestId || 'container'}
      {...(props as React.ComponentProps<typeof Atoms.Container>)}
    >
      {children}
    </Atoms.Container>
  );
}
