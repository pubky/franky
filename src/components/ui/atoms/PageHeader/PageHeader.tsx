import { cn } from '@/libs';

export function PageHeader({ children, className, ...props }: React.HTMLAttributes<HTMLDivElement>) {
  return (
    <div className={cn('flex flex-col gap-3', className)} {...props}>
      {children}
    </div>
  );
}
