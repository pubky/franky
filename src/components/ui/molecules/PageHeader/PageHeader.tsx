import { ReactNode } from 'react';
import { PageTitle, PageSubtitle } from '@/components/ui';
import { cn } from '@/libs';

interface PageHeaderProps {
  title: ReactNode;
  subtitle?: ReactNode;
  className?: string;
  titleSize?: 'medium' | 'large';
}

export function PageHeader({ title, subtitle, className, titleSize = 'large' }: PageHeaderProps) {
  return (
    <div className={cn('flex flex-col gap-3', className)}>
      <PageTitle size={titleSize}>{title}</PageTitle>
      {subtitle && <PageSubtitle>{subtitle}</PageSubtitle>}
    </div>
  );
}
