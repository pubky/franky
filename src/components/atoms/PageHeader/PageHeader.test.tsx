import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { PageHeader } from './PageHeader';

describe('PageHeader - Snapshots', () => {
  it('matches snapshot with default props', () => {
    const { container } = render(<PageHeader>Header content</PageHeader>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for different configurations', () => {
    const { container: defaultContainer } = render(<PageHeader>Default header</PageHeader>);
    expect(defaultContainer.firstChild).toMatchSnapshot();

    const { container: customClassContainer } = render(
      <PageHeader className="custom-header">Custom header</PageHeader>,
    );
    expect(customClassContainer.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for different children types', () => {
    const { container: simpleContainer } = render(<PageHeader>Simple content</PageHeader>);
    expect(simpleContainer.firstChild).toMatchSnapshot();

    const { container: complexContainer } = render(
      <PageHeader>
        <h1>Page Title</h1>
        <p>Page Subtitle</p>
      </PageHeader>,
    );
    expect(complexContainer.firstChild).toMatchSnapshot();

    const { container: nestedContainer } = render(
      <PageHeader>
        <div>
          <span>
            Complex <strong>Title</strong>
          </span>
          <span>
            Complex <em>Subtitle</em>
          </span>
        </div>
      </PageHeader>,
    );
    expect(nestedContainer.firstChild).toMatchSnapshot();
  });

  it('matches snapshots for additional props', () => {
    const { container: withIdContainer } = render(<PageHeader id="header-id">Header with ID</PageHeader>);
    expect(withIdContainer.firstChild).toMatchSnapshot();

    const { container: withDataTestIdContainer } = render(
      <PageHeader data-testid="page-header">Header with test ID</PageHeader>,
    );
    expect(withDataTestIdContainer.firstChild).toMatchSnapshot();
  });
});
