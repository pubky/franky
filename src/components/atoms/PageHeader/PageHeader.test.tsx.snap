// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageHeader - Snapshots > matches snapshot with complex children 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="container"
>
  <h1>
    Page Title
  </h1>
  <p>
    Page Subtitle
  </p>
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with custom className 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2 custom-header"
  data-testid="container"
>
  Custom header
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with data-testid prop 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="page-header"
>
  Header with test ID
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with default props 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="container"
>
  Header content
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with id prop 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="container"
  id="header-id"
>
  Header with ID
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with nested children 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="container"
>
  <div>
    <span>
      Complex 
      <strong>
        Title
      </strong>
    </span>
    <span>
      Complex 
      <em>
        Subtitle
      </em>
    </span>
  </div>
</div>
`;

exports[`PageHeader - Snapshots > matches snapshot with simple content 1`] = `
<div
  class="mx-auto w-full flex-col flex gap-3 pb-6 pt-2"
  data-testid="container"
>
  Simple content
</div>
`;
